api/authentication/views.py

```diff
@@ -361,3 +345,64 @@ class UserInactiveAPIView(APIView):
     
     def post(self, request, **kwargs):
         return Response({"Account inactive!"}, status=status.HTTP_400_BAD_REQUEST)
+
+class SetModeratorAPIView(APIView):
+    permission_classes = [AllowAny, ]
+    queryset = User.objects.none()
+    
+    @permission('IsAdminUser') 
+    def post(self, request, *args, **kwargs):
+        self.serializer_class = SetModeratorSerializer
```

Comment:
if you use `serializer_class` please add it in mail (parent) class above and later in methods you can call only `self.serializer_class`.  You have in this place like rewrite main option but you didn't added it in parent class.